src/Core/Geometry/StandardAttribNames.hpp

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include <Core/RaCore.hpp>
+
+namespace Ra {
+namespace Core {
+namespace Geometry {
+
+/// List of standard vertex attributes.
+/// Corresponding standard vertex attribute names are obtained with g_attribName
+/// Information which is in the mesh geometry
+enum MeshAttrib : uint {
+    VERTEX_POSITION,   ///< Vertex positions
+    VERTEX_NORMAL,     ///< Vertex normals
+    VERTEX_TANGENT,    ///< Vertex tangent 1
+    VERTEX_BITANGENT,  ///< Vertex tangent 2
+    VERTEX_TEXCOORD,   ///< U,V  texture coords (last coordinate not used)
+    VERTEX_COLOR,      ///< RGBA color.
+    VERTEX_WEIGHTS,    ///< Skinning weights (not used)
+    VERTEX_WEIGHT_IDX, ///< Associated weight bones
+
+    MAX_DATA
+};
+
+///@{
+/// Get the name expected for a given attrib.
+
+inline constexpr char const* getAttribName( MeshAttrib index ) {
+    constexpr char const* g_attribName[MeshAttrib::MAX_DATA] = { "in_position",
+                                                                 "in_normal",
+                                                                 "in_tangent",
+                                                                 "in_bitangent",
+                                                                 "in_texcoord",
+                                                                 "in_color",
+                                                                 "in_weight",
+                                                                 "in_weight_idx" };
+    return g_attribName[index];
+}
```

Comment:
It might be interesting to have also the reverse operation that convert a string to an enum.

Reply:
Well what is the use case ?

Reply:
for example for this line : if ( attr->getName() != std::string( "in_position" ) )
we can use std::unordered_map<std::string, MeshAttrib> string2meshAttrib to convert the name of attr with an hash operation like this
if (string2meshAttrib.at(attr->getName()) != MeshAttrib::VERTEX_POSITION)

Reply:
how it differs from
```cpp
attr->getName() != getAttrName(MeshAttrib::VERTEX_POSITION)
```

Reply:
the getAttName create an implicit std::string to compare with the other std::string
the other method do not created object

Reply:
Sorry I'm wrong, there is an std::operator == compare with an const char* in the stl

Reply:
> Well what is the use case ?

@dlyr : bijective function, introspection, serialization/deserialization, code generation ... do you need more ? 10 lines more will not hurt this PR but can serve many times.

@hiergaut : if the enum is an `enum class MeshAttrib : uint` instead of just `enum MeshAttrib :uint` (to prevent implicit cast from int to enum and enum to int), we can have an `operator==(std::string, MeshAttrib)` and `operator==(MeshAttrib, std::string,)` allowing to write 
```
MeshAttrib attrib = ....;
if ( attrib == "in_position" ) { ... }
```
which is simpler and easier to write/read than the explicit call to the unordered map.

Reply:
@MathiasPaulin could you give me an example code usage ?

Reply:
@MathiasPaulin This method is already present, and do not need enum to string conversion

Reply:
The purpose of StandardAttribNames is to store the convention used to name some attribs when loading files. Once name found, there is no need to use enum values (they are not stored in mesh, nor used by engine.

Reply:
> @MathiasPaulin This method is already present, and do not need enum to string conversion

@dlyr enum to string no, you're right but it should benefit from string to enum (the code do this). That's why i suggested.

Reply:
the code do string to enum, so I don't understand your request and how it's "the reverse operation".

Reply:
If I understand well, you suggest automatic "enum value to string" (or the other way round) for initialization is not what we want here.
The purpose of MeshAttrib is to have generic name we can edit if needed. 
The purpose of this PR is to refactor this existing behavior. 
We can discuss automatic generation of the string as done for instance for keymappingmananger or using magic enum. But it's out of scope of the PR.

Reply:
If fine, please resolve the conversation.

Reply:
I did some performance testing of the different methods proposed above, it seems that one method remains better than the others and offers an easier use for an user and developer.
[compareTest.zip](https://github.com/STORM-IRIT/Radium-Engine/files/8496468/compareTest.zip)

Reply:
[compareTest_v2.zip](https://github.com/STORM-IRIT/Radium-Engine/files/8497060/compareTest_v2.zip)
with magic_enum

Reply:
[compareTest.zip](https://github.com/STORM-IRIT/Radium-Engine/files/8498271/compareTest.zip)

Reply:
[compareTest.zip](https://github.com/STORM-IRIT/Radium-Engine/files/8498307/compareTest.zip)

Reply:
Thanks for the benchmarks I found quite instructive.
@dlyr what I understand from the code ```constexpr char const* getAttribName( MeshAttrib index )``` is that it does enum to string (eg : `VERTEX_POSITION` --> `"in_position"`) . In my suggestion of extension, what I call the reverse op, i.e. string to enum, is to have ```constexpr MeshAttrib getAttribIndex( char const* attribName )``` that does string to enum (eg : `"in_position"` --> `VERTEX_POSITION`). 
And yes, this is only for "standard" names/attribs.
The function `AttribArrayGeometry::getAttribHandle` I gave as an example of who can benefit from this (and any others that need to translate an attrib name to an attrib index)  will just call such an operator.

That being said, if you really think this suggestion is useless or that I totally misunderstood what is proposed in this PR (which just move existing things between radium components but also rewrite a little to improve performances and usage), you can also resolve this discussion. 

For me, it is still not clear why we do not need a string to enum and then why my suggestion is useless.

Reply:
Thanks for the clarification, Actually attribHandles and indices are not related to enum values. So I close the issue.